MapboxSceneKit/Math.swift

```diff
@@ -40,12 +42,27 @@ internal class Math {
         let lon = Double(x) / pow(2.0, Double(z)) * 360.0 - 180.0
         let n = Double.pi - 2.0 * Double.pi * Double(y) / pow(2.0, Double(z))
         let lat = Math.radiansToDegrees(atan(0.5 * (exp(n) - exp(-n))))
+        
         return (lat, lon)
     }
 
     static func tile2BoundingBox(x: Int, y: Int, z: Int) -> (latBounds: (CLLocationDegrees, CLLocationDegrees), lonBounds: (CLLocationDegrees, CLLocationDegrees)) {
         let topLeft = tile2LatLng(x: x, y: y, z: z)
         let bottomRight = tile2LatLng(x: x + 1, y: y + 1, z: z)
+        
         return ((bottomRight.lat, topLeft.lat), (topLeft.lon, bottomRight.lon))
     }
+    
+    static func zoomLevelForBounds(southWestCorner: CLLocation, northEastCorner: CLLocation) -> Int {
+        let distance = northEastCorner.distance(from: southWestCorner) / 1000.0 //use kilometers
+        let imageSize = Double(Constants.maxTextureImageSize)
+        let latitudeAdjustment = cos(.pi * northEastCorner.coordinate.latitude / 180)
+        let arg = Constants.earthDiameterInKilometers
```

Comment:
@avi-c: @jim-martin mentioned that you had gotten this formula from... somewhere. Do you happen to remember where? Mostly asking so we can think of a better name than "arg"